### PR TITLE
test/system: Add support for building FreeBSD testimages

### PR DESCRIPTION
**Pull request by dfr**
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/system/build-testimage [hunk 1]
```diff
@@ -12,6 +12,76 @@
...
+        cat <<EOF
+FROM ${busybox}
+ADD testimage-id pause /home/podman/
+RUN mkdir /usr
+RUN mkdir /srv
+EOF
+    }
+
+    function repo_for_arch {
+        echo dougrabson
```
Comment (rhatdan) [inline test/system/build-testimage:81]: debugging?

Comment (dfr) [reply test/system/build-testimage:81]: Not really. On linux, this builds a multi-arch manifest from single-arch images located at e.g. docker.io/amd64/busybox. For freebsd, my busybox image at quay.io/dougrabson/busybox is a multi-arch manifest so we can use the same image name for both amd64 and arm64 in the code below.

File: test/system/build-testimage [hunk 2]
```diff
@@ -12,6 +12,76 @@
...
+
+    function busyboximg {
+        echo quay.io/dougrabson/busybox:1.36.1
+    }
+
+    function build_containerfile {
+        # busybox because it's small and has a command set which is close to what
+        # the test suite needs.
+        cat <<EOF
+FROM ${busybox}
```
Comment (edsantiago) [inline test/system/build-testimage:73]: This is the crux; this is why I feel obligated to block this. This and line 51, but mostly this.

The source image we use for `testimage`, `docker.io/ARCH/alpine`, is not under our control but is presumably a known quantity. This one is a black box. It is presumably a busybox image that you have crafted yourself, either from scratch (my guess) or based on an existing container image from somewhere else. It's basically magic.

Which is OK for an image used only by you, ... but this is not something that we (upstream) can in any way test, verify, maintain, or support. This is basically asking upstream to carry a very large complicated set of patches for something that upstream can't possibly work with. 

I'm really sorry. I can't approve this.

**Review by edsantiago — CHANGES_REQUESTED** [review]
@dfr thank you for your PR and thank you for your patience. I am sorry to get to this so late. I am also sorry that I can't approve this; see explanation inline.

In a world where I could see myself approving something like this, I would require:
* A robust source for the source image (documented, reproducible, secure);
* An `if freebsd` override, not an `if linux / elif freebsd`. The 99% case should be given prominent weight for maintainability, the 1% should be the conditional.
** Possibly done in a separate script file (`source build-testimage.$(uname -s)`) to reduce maintenance load on the 99% case
* mode reset back to 755 (I assume this was an accident)
* some understanding of how the resulting images will be used, and what other changes would be needed to the test suite, because I foresee hundreds of `skip_if_freebsd` lines in anything having to do with cgroups, SELinux, and probably much more.

But mostly it's the source image uncertainty.

I'm really sorry. I'm really thankful for your work, but I just can't see it in `main` as it is.

**Comment by dfr** [comment]
The mode change was unintended - I messed it up while expanding unwanted tabs.

I agree with you on the importance of having a clear provenance on default images. The busybox image itself is simple:
```
FROM quay.io/dougrabson/freebsd-minimal:13.2 as builder
COPY FreeBSD.conf /usr/local/etc/pkg/repos/FreeBSD.conf
RUN pkg update
RUN pkg install -y busybox
RUN /etc/rc.d/os-release start

FROM scratch
COPY --from=builder /etc /etc
COPY --from=builder /var/run /var/run
COPY --from=builder /usr/local/bin/busybox /bin/busybox
COPY --from=builder /libexec/ld-elf.so.1 /libexec/ld-elf.so.1
COPY --from=builder /lib/libcrypt.so.5 /lib/libcrypt.so.5
COPY --from=builder /lib/libm.so.5 /lib/libm.so.5
COPY --from=builder /lib/libc.so.7 /lib/libc.so.7
RUN ["busybox", "sh", "-c", "for i in $(busybox --list); do busybox ln -sf /bin/busybox /bin/$i; done"]
RUN ["mkdir", "-m", "1777", "/tmp"]
```
but that doesn't help because there is no reason to trust ```quay.io/dougrabson/freebsd-minimal:13.2``` either other than my statement that I built it.  I am working within the FreeBSD project to find a way to have a set of trusted FreeBSD images but right now, the release engineer's bandwidth is better spent on the upcoming FreeBSD 14.0 release.

Lets park the idea of building a project-owned freebsd testimage for now. I would like to try to merge at least some of my changes to the test suite itself - I have most tests passing up to 120-load.bats and at the current count, 27 calls to skip_if_freebsd, three of which are reminders to me to debug something.

**Comment by edsantiago** [comment]
Thank you for understanding! Yes, a project-owned image would go a long ways toward moving this along! I look forward to reviewing this again. And ITM I know that we all appreciate your frequent fixes to our test code.

**Pull request closed without merging**